monitoring/src/check-page/tcfv2.ts

```diff
@@ -94,3 +99,287 @@ export const checkPage = async function (
+	log_info(`Clicking on "Yes I'm Happy" on CMP`);
+	const frame = page
+		.frames()
+		.find((f) => f.url().startsWith('https://sourcepoint.theguardian.com'));
+	if (frame === undefined) {
+		return;
+	}
+
+	await frame.click('button[title="Yes, I’m happy"]');
+};
+
+const checkCMPIsOnPage = async (page: Page) => {
+	log_info(`Waiting for CMP: Start`);
+	await page.waitForSelector('[id*="sp_message_container"]');
+	log_info(`Waiting for CMP: Finish`);
+};
+
+const checkCMPIsNotVisible = async (page: Page) => {
+	log_info(`Checking CMP is Hidden: Start`);
+
+	const getSpMessageDisplayProperty = function () {
+		const element = document.querySelector('[id*="sp_message_container"]');
+		if (element) {
+			const computedStyle = window.getComputedStyle(element);
+			return computedStyle.getPropertyValue('display');
+		}
+	};
+
+	const display = await page.evaluate(getSpMessageDisplayProperty);
+
+	// Use `!=` rather than `!==` here because display is a DOMString type
+	if (display && display != 'none') {
+		throw Error('CMP still present on page');
+	}
+
+	log_info('CMP hidden or removed from page');
+};
+
+const checkCMPDidNotLoad = async (page: Page) => {
+	log_info(`Checking CMP does not load: Start`);
+
+	const spMessageContainer = await page.$('[id*="sp_message_container"]');
+
+	if (spMessageContainer !== null) {
+		log_error(`Checking CMP does not load: Failed`);
+		throw Error('CMP present on page');
+	}
+
+	log_info(`Checking CMP does not load: Complete`);
+};
+
+const reloadPage = async (page: Page) => {
+	log_info(`Reloading page: Start`);
+	const reloadResponse = await page.reload({
+		waitUntil: ['networkidle0', 'domcontentloaded'],
+		timeout: 30000,
+	});
+	if (!reloadResponse) {
+		log_error(`Reloading page: Failed`);
+		throw 'Failed to refresh page!';
+	}
+	log_info(`Reloading page: Complete`);
+};
+
+/**
+ * Checks that ads load correctly for the second page a user goes to
+ * when visiting the site, with respect to and interaction with the CMP.
+ */
+
+const checkSubsequentPage = async (url: string) => {
+	// const page = await synthetics.getPage();
+	const browser: Browser = await makeNewBrowser();
+	const page: Page = await browser.newPage();
+
+	log_info(`Start checking subsequent Page URL: ${url}`);
+
+	await loadPage(page, url);
+
+	// There is no CMP since this we have already accepted this on a previous page.
+	await checkTopAdHasLoaded(page);
+
+	const client = await page.target().createCDPSession();
+	await clearCookies(client);
+
+	await reloadPage(page);
+
+	await checkTopAdDidNotLoad(page);
+
+	await interactWithCMP(page);
+
+	await checkCMPIsNotVisible(page);
+
+	await checkTopAdHasLoaded(page);
+};
+
+const loadPage = async (page: Page, url: string) => {
+	log_info(`Loading page: Start`);
+
+	const response = await page.goto(url, {
+		waitUntil: 'domcontentloaded',
+		timeout: 30000,
+	});
+
+	// For some reason VSCode thinks the conditional is not needed, because `!response` is always falsy 🤔
```

Comment:
I came across this same issue when adapting this code the first time (see `checkPage_old`). I think you can safely remove this statement. `page.goto` will throw if there is an issue rather than return something falsy.